### ISSUE TITLE
Separate diagnostic category for argument unpacking vs regular argument type errors

### ISSUE THREAD

**Issue by smojef** [microsoft/pyright#10711]
### Summary
This is a request to split reportArgumentType into separate diagnostic categories to allow disabling false positives from dictionary/sequence unpacking while preserving valuable type checking for regular argument mismatches.

### Problem
Currently, `reportArgumentType` covers both:
1. *Legitimate type errors* (should be reported)
2. *False positives from unpacking* (should be ignorable)

This forces developers to either:
- Disable all argument type checking (losing valuable safety)
- Live with numerous false positives (poor developer experience)
- Add `# type: ignore` comments everywhere (maintenance burden)

### Examples
```python
def func_with_catch_all(file:str, title='Default Title', **cfg):
    print(locals())

cfg_kw = dict(line_size=2, marker_size=10)
func_with_catch_all('data.txt', **cfg_kw)
""" TO IGNORE:
Argument of type "int" cannot be assigned to parameter "title" of type "str" in function "func_with_catch_all"
"""
func_with_catch_all([1, 2, 3], 'Custom Title', **cfg_kw)
""" TO KEEP:
Argument of type "list[int]" cannot be assigned to parameter "file" of type "str" in function "func_with_catch_all"
"""

def func_with_many_args(x, y, marker_size=5, marker_color="#131212", line_size=1, line_color='#ffffff'):
    print(locals())

marker_kw = dict(marker_size=10, marker_color='#ff0000')
line_kw = dict(line_size=2, line_color='#0000ff')
func_with_many_args(1, 2, **line_kw, **marker_kw)
""" TO IGNORE:
Argument of type "int | str" cannot be assigned to parameter "marker_size" of type "int" in function "func_with_many_args"
  Type "int | str" is not assignable to type "int"
Argument of type "int | str" cannot be assigned to parameter "marker_color" of type "str" in function "func_with_many_args"
  Type "int | str" is not assignable to type "str"
Argument of type "int | str" cannot be assigned to parameter "line_size" of type "int" in function "func_with_many_args"
  Type "int | str" is not assignable to type "int"
Argument of type "int | str" cannot be assigned to parameter "line_color" of type "str" in function "func_with_many_args"
  Type "int | str" is not assignable to type "str"
"""
func_with_many_args(1, 2, 3, 4, 5, 6)
""" TO KEEP:
Argument of type "Literal[4]" cannot be assigned to parameter "marker_color" of type "str" in function "func_with_many_args"
"""

def func_with_ranges(x:float, in_lo:float, in_hi:float, out_lo=0, out_hi=1, clip=True):
    print(locals())

in_range = [10, 20]
out_range = [-1, +1]
func_with_ranges(123.4, *in_range, *out_range)
""" TO IGNORE:
Argument of type "int" cannot be assigned to parameter "clip" of type "bool" in function "func_with_ranges"
"""
pct_range = (0, 100)
func_with_ranges(123.4, *in_range, *pct_range)
""" TO IGNORE:
Expected 6 positional arguments
"""
func_with_ranges(123.4, (10, 20), [-1, +1])
""" TO KEEP:
Argument of type "tuple[Literal[10], Literal[20]]" cannot be assigned to parameter "in_lo" of type "float" in function "func_with_ranges"
"""
```
The same behaviour also occurrs on the current [pyright playground](https://pyright-play.net/?pyrightVersion=1.1.400&pythonVersion=3.14&code=CYUwZgBGCuB2DGB9A7gSwC4AtHwIbvm1wBtiAKMVYkALgGd0AnAGgnQ2oF4ByAEXFzRi6CABUOIbqwBU0%2BGADmAShoAoCBogAHRqljoyxAPZ5idMkqWrV8hYgDWyCJwjBU8A8T0hEdVAC8QTgAmVgBbXEZ7EEZfAKCARgAGKxgEFAxsPAIiUjJuYHxcADp0AA90KQhZWwdkJQgAYjEAeQgASQBxADkWgCUAUVU0pDQsHHxCRBJyAG0E1lCIAGYAXVZuAGFoBiMwsQkqmsU6hubRNoBpAYGABWtQSBGM8YjYAE9pxgVzMtZ38KRaKxPyBTgAVkBURiOCMxkYnAARI0EssEsF0YjWF5YD5QYlsd5YfCeI0wOSKdwVOpNDo9J4TCRzJZrBFobFHM5XO4DGzgXEwckofz4HCjAjuGSwEkZUkqaocT5OS43B5DET8SFCbjieLSbKkuT5c8xtg3p9Ij8yAsIEtZIq6jJpHyYY4zq0Oj1%2BkMTZlEOavlabUtlqwACysSEQABs7ouEGudwe4CgcFGfsYuFgChAvxoYGM%2BFYekQxnzhfQxdgiEwqHLRiLECM0HQpaMnCSrGbrdrnBt8C8Wk4okY0BA1M02l0%2BkMjLMFisqhLmezIC5807tqSq1U3cQK5z64AtDaANQJHe%2B8YH3PW4LLYoR6rLrM5mR7m9xtpdXqDVRaDx91fNcXDITdkhSYY0xebAb3MdEHyfaQX1XGQANbT8mg9H9vSg9JTSA1d4PvR9WGtTdghSVhZhPVhz1WL8Exue4gA).

### Proposed Solution
Split `reportArgumentType` into:
- `reportArgumentType` - Regular argument type mismatches (keep current behavior)
- `reportArgumentTypeUnpacking` - Type errors from **dict and *sequence unpacking

### Impact
This would allow developers to:
- Keep valuable type checking for regular arguments
- Disable false positives from unpacking operations
- Use common Python patterns without type checker friction

**Comment by erictraut** [comment]
Thanks for the suggestion. 

We have split diagnostic categories on occasion in the past, but this is a breaking change for existing pyright users, so the bar is quite high. I don't think the justification for this request is strong enough to justify the potential churn this would cause for pyright users.

I'm willing to revisit this decision in the future if there's strong enough signal from other pyright users (in the form of upvotes on this issue).
